src/database/database-sqlite3.cpp

```diff
@@ -779,3 +779,105 @@ void AuthDatabaseSQLite3::writePrivileges(const AuthEntry &authEntry)
 		sqlite3_reset(m_stmt_write_privs);
 	}
 }
+
+ModMetadataDatabaseSQLite3::ModMetadataDatabaseSQLite3(const std::string &savedir):
+	Database_SQLite3(savedir, "mod_storage"), ModMetadataDatabase()
+{
+}
+
+ModMetadataDatabaseSQLite3::~ModMetadataDatabaseSQLite3()
+{
+	FINALIZE_STATEMENT(m_stmt_remove)
+	FINALIZE_STATEMENT(m_stmt_set)
+	FINALIZE_STATEMENT(m_stmt_get)
+}
+
+void ModMetadataDatabaseSQLite3::createDatabase()
+{
+	assert(m_database); // Pre-condition
+
+	SQLOK(sqlite3_exec(m_database,
+		"CREATE TABLE IF NOT EXISTS `pairs` (\n"
+			"	`modname` TEXT NOT NULL,\n"
```

Comment:
This should be a `VARCHAR(30)` or so, does Minetest have a set limit on mod names?

Reply:
Apparently SQLite treats `VARCHAR(N)` exactly the same as `TEXT` (it doesn't enforce the limit.) I could just change the type `VARCHAR(64)` or something.

Reply:
> This should be a `VARCHAR(30)` or so, does Minetest have a set limit on mod names?

It should not be a `VARCHAR(30)` for (1) future flexibility (who will remember to change this if the mod name length limit is bumped? a constant would have be introduced) and (2) SQLite ignores the type arguments anyways:
> Note that numeric arguments in parentheses that following the type name (ex: "VARCHAR(255)") are ignored by SQLite - SQLite does not impose any length restrictions (other than the large global SQLITE_MAX_LENGTH limit) on the length of strings, BLOBs or numeric values.